packages/libs/components/src/map/SemanticMarkers.tsx

```diff
@@ -124,8 +127,18 @@ export default function SemanticMarkers({
         setConsolidatedMarkers(recenteredMarkers);
       }
 
-      // Update previous markers
-      setPrevMarkers(recenteredMarkers);
+      // To prevent infinite loops, especially when in "other worlds",
+      // update previous markers unless they are deep-equals.
+      // Only check the props - hopefully more efficient!
+      // If there are any function props (there aren't right now) these
+      // will be compared with referential-equals.
+      if (
+        !isEqual(
+          recenteredMarkers.map(({ props }) => props),
+          prevMarkers.map(({ props }) => props)
+        )
+      )
+        setPrevMarkers(recenteredMarkers);
```

Comment:
Can we rename `prevMarkers` to `prevRecenteredMarkers`, and `setPrevMarkers` to `setPrevRecenteredMarkers`?

Reply:
Sure